library/core/src/ops/index_range.rs

```diff
@@ -45,7 +45,8 @@ impl IndexRange {
     #[inline]
     pub const fn len(&self) -> usize {
         // SAFETY: By invariant, this cannot wrap
-        unsafe { self.end.unchecked_sub(self.start) }
+        // Using the intrinsic avoids a superfluous UB check.
+        unsafe { crate::intrinsics::unchecked_sub(self.end, self.start) }
```

Comment:
I think the other changes in here are justified, but not this one, because it's relying on non-local enforcement. I'd much rather see LLVM figure this one out. Or at the very least, instead of using the usual "superfluous UB check" comment, say something like "Having a UB check here impedes LLVM optimizations" and reference the issue.

Reply:
Well this is the main one that's involved in that particular test case, and surely more common in general, so I'd prefer to keep it. I'll update the comment as you suggest.

The other two cases are just for iterator `advance`/`nth` methods.